core/jvm/src/test/scala/scalaz/zio/FiberSpec.scala

```diff
@@ -11,7 +13,7 @@ class FiberSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRun
     for {
       ref   <- Ref.make(false)
       fiber <- IO.unit.bracket(_ => ref.set(true))(_ => IO.never).fork
-      _     <- fiber.toManaged.use(_ => IO.unit)
+      _     <- fiber.toManaged.use(_ => clock.sleep(20.millis))
```

Comment:
To make the test deterministic and run as fast as possible, let's instead just join the fiber after we use it:


```scala
for {
       ref   <- Ref.make(false)		       ref   <- Ref.make(false)
       fiber <- IO.unit.bracket(_ => ref.set(true))(_ => IO.never).fork		       fiber <- IO.unit.bracket(_ => ref.set(true))(_ => IO.never).fork
       _     <- fiber.toManaged.use(_ => IO.unit)
		       _     <- fiber.await
```

Reply:
Hmm, actually, maybe I'm wrong here, shouldn't `toManaged` already do that?

Reply:
we can add `clock.sleep`before we call `toManaged` because the execution of IO.unit is faster and the fiber will be interrupted before updating the value of ref

Reply:
Ah, `toManaged` interrupts the fiber after `use`. So we need a latch to make test deterministic and fast.

Here's what we need, I think:

```scala
for {
       ref   <- Ref.make(false)
       latch <- Promise.make[Nothing, Unit]
       fiber <- IO.unit.bracket(_ => ref.set(true) *> latch.succeed(()))(_ => IO.never).fork
       _     <- fiber.toManaged.use(_ => latch.await)
```

Reply:
@wi101 The sleep won't be reliable and will delay test. So let's try to use a latch here if we can. See above code snippet.

Reply:
@jdegoes 👍 will do that